etl/dags/extract_weather_data.py

```diff
@@ -0,0 +1,94 @@
+from datetime import timedelta, datetime
+from airflow import DAG
+from airflow.operators.python_operator import PythonOperator
+from airflow.utils.dates import days_ago
+import json
+from google.cloud.storage import Client
+import requests
+from deepmerge import always_merger
+import os
+
+BASE_URL = "http://api.weatherapi.com/v1/history.json?key={}&q={}+united+states&dt={}"
+# Get a weatherapi.com api key
+API_KEY = os.environ.get("API_WEATHER_KEY", "a70a4e2736644cdcb9d85348202404")
```

Comment:
Keys must never be committed to version control. This must be removed and the history squashed before merging. We can register the API key as an airflow secret.